test/extended/util/configv1shim_test.go

```diff
@@ -0,0 +1,580 @@
+			},
+			expectedWatchEvents: []watch.Event{
+				{Type: watch.Added, Object: createNetworkObject("staticObject")},
+				{Type: watch.Added, Object: createNetworkObject("realObject")},
+				{Type: watch.Added, Object: createNetworkObject("realObject2")},
+			},
+		},
+		{
+			name: "merging static and real network objects, static object override",
+			staticObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			realObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+				createNetworkObject("realObject"),
+			},
+			watch: func(client *ConfigClientShim, ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+				return client.ConfigV1().Networks().Watch(ctx, opts)
+			},
+			expectedWatchEvents: []watch.Event{
+				{Type: watch.Added, Object: createNetworkObject("staticObject")},
+				{Type: watch.Added, Object: createNetworkObject("realObject")},
+			},
+		},
+		{
+			name: "merging static and real network objects, field selector match",
+			staticObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			realObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			watch: func(client *ConfigClientShim, ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+				return client.ConfigV1().Networks().Watch(ctx, opts)
+			},
+			fieldSelector: "metadata.name==staticObject",
+			expectedWatchEvents: []watch.Event{
+				{Type: watch.Added, Object: createNetworkObject("staticObject")},
+			},
+		},
+		{
+			name: "merging static and real network objects, field selector no match",
+			staticObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			realObjects: []runtime.Object{
+				createNetworkObject("staticObject"),
+			},
+			watch: func(client *ConfigClientShim, ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+				return client.ConfigV1().Networks().Watch(ctx, opts)
+			},
+			fieldSelector:       "metadata.name=!staticObject",
+			expectedWatchEvents: []watch.Event{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			configClient := fakeconfigv1client.NewSimpleClientset()
+
+			client := NewConfigClientShim(
+				configClient,
+				test.staticObjects,
+			)
+
+			// The watch request has to created first when using the fake clientset
+			resultChan, err := test.watch(client, context.TODO(), metav1.ListOptions{FieldSelector: test.fieldSelector})
+			if err != nil {
+				t.Fatalf("Expected no error, got %q instead", err)
+			}
+			defer resultChan.Stop()
+
+			// And only then the fake clientset can be populated to generate the watch event
+			for _, obj := range test.realObjects {
+				configClient.Tracker().Add(obj)
+			}
+
+			// verify the watch events
+			ticker := time.NewTicker(500 * time.Millisecond)
+			size := len(test.expectedWatchEvents)
+			eventsSize := len(resultChan.ResultChan())
```

Comment:
this seems potentially racy, because of the processing loop. Maybe we can check that the len of the channel is 0 after the loop instead?

Reply:
Checking if there's another event in the channel at the end of the test